src/webgpu/api/operation/rendering/blending.spec.ts

```diff
@@ -7,7 +8,294 @@ TODO:
 - ?
 `;
 
+import { params, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+const kBlendFactors: GPUBlendFactor[] = [
+  'zero',
+  'one',
+  'src-color',
+  'one-minus-src-color',
+  'src-alpha',
+  'one-minus-src-alpha',
+  'dst-color',
+  'one-minus-dst-color',
+  'dst-alpha',
+  'one-minus-dst-alpha',
+  'src-alpha-saturated',
+  'blend-color',
+  'one-minus-blend-color',
+];
+
+const kBlendOperations: GPUBlendOperation[] = [
+  'add', //
+  'subtract',
+  'reverse-subtract',
+  'min',
+  'max',
+];
+
+function mapColor(
+  col: GPUColorDict,
+  f: (v: number, k: keyof GPUColorDict) => number
+): GPUColorDict {
+  return {
+    r: f(col.r, 'r'),
+    g: f(col.g, 'g'),
+    b: f(col.b, 'b'),
+    a: f(col.a, 'a'),
+  };
+}
+
+function computeBlendFactor(
+  src: GPUColorDict,
+  dst: GPUColorDict,
+  blendColor: GPUColorDict | undefined,
+  factor: GPUBlendFactor
+): GPUColorDict {
+  switch (factor) {
+    case 'zero':
+      return { r: 0, g: 0, b: 0, a: 0 };
+    case 'one':
+      return { r: 1, g: 1, b: 1, a: 1 };
+    case 'src-color':
+      return { ...src };
+    case 'one-minus-src-color':
+      return mapColor(src, v => 1 - v);
+    case 'src-alpha':
+      return mapColor(src, () => src.a);
+    case 'one-minus-src-alpha':
+      return mapColor(src, () => 1 - src.a);
+    case 'dst-color':
+      return { ...dst };
+    case 'one-minus-dst-color':
+      return mapColor(dst, v => 1 - v);
+    case 'dst-alpha':
+      return mapColor(dst, () => dst.a);
+    case 'one-minus-dst-alpha':
+      return mapColor(dst, () => 1 - dst.a);
+    case 'src-alpha-saturated': {
+      const f = Math.min(src.a, 1 - dst.a);
+      return { r: f, g: f, b: f, a: 1 };
+    }
+    case 'blend-color':
+      assert(blendColor !== undefined);
+      return { ...blendColor };
+    case 'one-minus-blend-color':
+      assert(blendColor !== undefined);
+      return mapColor(blendColor, v => 1 - v);
+  }
+}
+
+function computeBlendOperation(src: GPUColorDict, srcFactor: GPUColorDict,
+  dst: GPUColorDict, dstFactor: GPUColorDict,operation: GPUBlendOperation) {
```

Comment:
Confused why the linter is not forcing this to get autoformatted. Something must have broken in the CI at some point...